src/engine/plugins/viaroute.cpp

```diff
@@ -82,21 +82,46 @@ Status ViaRoutePlugin::HandleRequest(const std::shared_ptr<datafacade::BaseDataF
                 !continue_straight_at_waypoint;
         }
     };
-    util::for_each_pair(snapped_phantoms, build_phantom_pairs);
 
-    if (1 == raw_route.segment_end_coordinates.size())
+    if (phantom_node_pairs.size() == 2)
```

Comment:
Yeah. kind of doesn't help next to `const` functions. Can save you from `if( var = 2 )` errors, though.